tests_onehotencoder.py

```diff
@@ -0,0 +1,236 @@
+dense1_1h_minimum_fraction = np.array([[0, 1, 0, 1, 1],
+                                       [0, 1, 1, 0, 1],
+                                       [1, 0, 0, 1, 1]])
+
+# Including NaNs
+dense2 = np.array([[0, np.NaN, 0],
+                   [np.NaN, 0, 2],
+                   [1, 1, 1],
+                   [np.NaN, 0, 1]])
+dense2_1h = np.array([[0, 1, 0, 1, 0, 0, 1, 0, 0],
+                      [1, 0, 0, 0, 1, 0, 0, 0, 1],
+                      [0, 0, 1, 0, 0, 1, 0, 1, 0],
+                      [1, 0, 0, 0, 1, 0, 0, 1, 0]])
+
+dense2_1h_minimum_fraction = np.array([[1, 0, 1, 0, 1, 0],
+                                       [0, 1, 0, 1, 1, 0],
+                                       [1, 0, 1, 0, 0, 1],
+                                       [0, 1, 0, 1, 0, 1]])
+
+dense2_partial_1h = np.array([[0., 1., 0., 1., 0., 0., 0.],
+                              [1., 0., 0., 0., 1., 0., 2.],
+                              [0., 0., 1., 0., 0., 1., 1.],
+                              [1., 0., 0., 0., 1., 0., 1.]])
+
+dense2_1h_minimum_fraction_as_sparse = np.array([[0, 0, 1, 0, 0, 0],
+                                                 [0, 1, 0, 0, 1, 0],
+                                                 [1, 0, 0, 1, 0, 1],
+                                                 [0, 1, 0, 0, 0, 1]])
+
+# All NaN slice
+dense3 = np.array([[0, 1, np.NaN],
+                   [1, 0, np.NaN]])
+dense3_1h = np.array([[1, 0, 0, 1, 1],
+                      [0, 1, 1, 0, 1]])
+
+sparse1 = scipy.sparse.csc_matrix(([3, 2, 1, 1, 2, 3],
+                                   ((1, 4, 5, 2, 3, 5),
+                                    (0, 0, 0, 1, 1, 1))), shape=(6, 2))
+sparse1_1h = scipy.sparse.csc_matrix(([1, 1, 1, 1, 1, 1],
+                                      ((5, 4, 1, 2, 3, 5),
+                                       (0, 1, 2, 3, 4, 5))), shape=(6, 6))
+sparse1_paratial_1h = scipy.sparse.csc_matrix(([1, 1, 1, 1, 2, 3],
+                                               ((5, 4, 1, 2, 3, 5),
+                                                (0, 1, 2, 3, 3, 3))),
+                                              shape=(6, 4))
+
+# All zeros slice
+sparse2 = scipy.sparse.csc_matrix(([2, 1, 0, 0, 0, 0],
+                                   ((1, 4, 5, 2, 3, 5),
+                                    (0, 0, 0, 1, 1, 1))), shape=(6, 2))
+sparse2_1h = scipy.sparse.csc_matrix(([1, 1, 1, 1, 1, 1],
+                                      ((5, 4, 1, 2, 3, 5),
+                                       (0, 1, 2, 3, 3, 3))), shape=(6, 4))
+
+sparse2_csr = scipy.sparse.csr_matrix(([2, 1, 0, 0, 0, 0],
+                                      ((1, 4, 5, 2, 3, 5),
+                                       (0, 0, 0, 1, 1, 1))), shape=(6, 2))
+sparse2_csr_1h = scipy.sparse.csr_matrix(([1, 1, 1, 1, 1, 1],
+                                         ((5, 4, 1, 2, 3, 5),
+                                          (0, 1, 2, 3, 3, 3))), shape=(6, 4))
+
+
+class OneHotEncoderTest(unittest.TestCase):
```

Comment:
For now, let's remove the class structure and indent the `test_` functions back.